limits/aio/storage/base.py

```diff
@@ -179,3 +179,57 @@ async def get_moving_window(
         :return: (start of window, number of acquired entries)
         """
         raise NotImplementedError
+
+
+class SlidingWindowCounterSupport(ABC):
+    """
+    Abstract base for storages that intend to support
+    the sliding window counter strategy
+    """
+
+    def __new__(cls, *args: Any, **kwargs: Any) -> SlidingWindowCounterSupport:  # type: ignore[misc]
+        inst = super().__new__(cls)
+
+        for method in {"acquire_sliding_window_entry", "get_sliding_window"}:
+            setattr(
+                inst,
+                method,
+                _wrap_errors(cast(Storage, inst), getattr(inst, method)),
+            )
+
+        return inst
+
+    @abstractmethod
+    async def acquire_sliding_window_entry(
+        self,
+        key: str,
+        limit: int,
+        expiry: int,
+        amount: int = 1,
+    ) -> bool:
+        """
+        Acquire an entry. Shift the current window to the previous window if it expired.
+        :param current_window_key: current window key
+        :param previous_window_key: previous window key
+        :param limit: amount of entries allowed
+        :param expiry: expiry of the entry
+        :param amount: the number of entries to acquire
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    async def get_sliding_window(
+        self, key: str, expiry: Optional[int] = None
```

Comment:
Could we make `expiry` required? (i.e. not `Optional`)

Reply:
Yes! It was not needed for some implementations before but now that we reworked them, it is needed for all. Plus, it removes a lot of dead code that was not covered by tests. Thanks